### ISSUE TITLE
when Chassis Serial is 00, bmcdiscover can not create correct node name

### ISSUE THREAD

**Issue by bybai** [xcat2/xcat-core#6406]
when Chassis Serial is 00, bmcdiscover can not create correct node name.

When Chassis Serial is 00:
bmcdiscover will go into bmcdiscover.pm 1139:
```
                    if (($fru_output =~ /Chassis Part Number\s*:\s*(\S*).*Chassis Serial\s*:\s*(\S*)/))
{
                        $mtm    = $1;
                        $serial = $2;
                        last;
                    }
```
Then, in bmcdiscover.pm line 1161:
```
                $serial = '' if ($serial =~ /^0+$/);
```
then, in bmcdiscover.pm line 1180:
```
if ($mtm and $serial) {
...
```
At last, the node name is empty

```
]# lsxcatd -v
Version 2.14.6 (git commit 7c787c499ebca7a98d6fcdb3577f701ac16d7ab3, built Fri Mar 29 03:19:08 EDT 2019)
```

**Comment by whowutwut** [comment]
Is this a valid serial number?  It's just `00`?

**Comment by gurevichmark** [comment]
Resolved by #6451
